ext/handlers/filters/message/message.go

```diff
@@ -47,6 +47,16 @@ func ForwardFromChatID(id int64) filters.Message {
 	}
 }
 
+func Regex(pattern string) filters.Message {
+	return func(m *gotgbot.Message) bool {
+		matched, err := regexp.MatchString(pattern, m.Text)
+		if err != nil {
+			return false
```

Comment:
This swallows all errors, making it very hard to debug. How do you know your match failed?

One option would be to change the definition to `Regex(p string) (filters.Message, error)`. This stops users from being able to instantiate the filter inline, but still allows them to handle the error.

Reply:
I see, going to try it out today. :) Thanks for the brief explanation.

Reply:
Why not changing the type `filters.Message` from `func(*gotgbot.Message) bool` to `func(*gotgbot.Message) (bool, error)`?
Isn't it a good idea?

Reply:
@anonyindian  please check my pull request [here](https://github.com/anonyindian/gotgbot/pull/1).

Reply:
The compile error only ever happens once (at compile time). The error can also only ever be handled at compile time, so there's no way to action any errors that would get returned with (bool, error) - so, what do you gain by returning the error?